pkg/kubelet/kubelet_pods.go

```diff
@@ -1728,8 +1743,11 @@ func hasHostVolume(pod *v1.Pod) bool {
 	return false
 }
 
-// hasHostNamespace returns true if hostIPC, hostNetwork, or hostPID are set to true.
+// hasHostNamespace returns true if hostIPC, hostNetwork, or hostPID are set to true or userNamespaceRemapping is set to false.
 func hasHostNamespace(pod *v1.Pod) bool {
+	if pod.Spec.HostUser != nil {
+		return pod.Spec.HostIPC || pod.Spec.HostNetwork || pod.Spec.HostPID || *pod.Spec.HostUser
+	}
 	if pod.Spec.SecurityContext == nil {
```

Comment:
why was `pod.Spec.SecurityContext` short-circuiting this previously and why is that no longer valid?

Reply:
this `pod.Spec.SecurityContext` is [left by mistake long time ago](https://github.com/kubernetes/kubernetes/commit/5e1adf91dfbe6ec9f96950e9b4e28523b8d3ccd8#diff-10055ae93a8699af13ceba0482fc43c3R1476). This check does not make sense IMO. Will remove this. Thanks!!